specification.md

```diff
@@ -1855,6 +1868,48 @@ for this connection.
 This field is optional: if omitted the client SHOULD NOT use a client-side certificate.
 This field can be used to perform a client certificate revocation/rotation.
 
+##### OpAMPConnectionSettings.heartbeat_interval_seconds
+
+If the ReportsHeartbeat capability is true, the Client MUST use the offered heartbeat
+interval to periodically send an AgentToServer message. If the capability is true
+and the Server sets heartbeat_interval_seconds to 0, Agent heartbeats should be disabled.
+At a minimum the instance_uid field MUST be set.
+An HTTP based-client MUST use the heartbeat interval as its polling interval.
+
+A heartbeat is used to keep a load balancer connection active and inform the server that
+the Agent is still alive and active. A server could use the heartbeat to make decisions about
+the liveness of the connected Agent.
+
+The flow for negotiating a heartbeat is described as so:
+
+```
+┌──────────┐                       ┌──────────┐
+│          │ (1) Connect           │          │
+│          ├──────────────────────►│          │
+│          │                       │          │
+│          │ (2) Set Heartbeat     │          │
+│          │◄──────────────────────┤          │
+│          │     Interval          │          │
+│          │                       │          │
+│  Agent   │ (3) Send Heartbeat    │  Server  │
+│          ├──────────────────────►│          │
+│          │                       │          │
+│          │  ... heartbeat        │          │
+│          │      interval         │          │
+│          │                       │          │
+│          │ (4) Send Heartbeat    │          │
+│          ├──────────────────────►│          │
+│          │                       │          │
+└──────────┘                       └──────────┘
+```
+
+1. The agent connects to the server and optionally sets the ReportsHeartbeat capability. If the Agent does NOT set this capability, no heartbeats will occur.
+2. If the Agent sets the ReportsHeartbeat capability, the server MUST respond with either a 0 indicating that heatbeats are disabled. Otherwise, the server will set a heartbeat_interval_seconds in the OpAMPConnectionSettings message. It is recommended for servers to set a 30s interval, if desired.
```

Comment:
```suggestion
2. If the Agent sets the ReportsHeartbeat capability, the server MUST respond by setting an interval in the heartbeat_interval_seconds field within the OpAMPConnectionSettings message. The value can either the desired interval, or `0`, indicating that the client should not send heartbeats. 30s is the recommended default interval.
```
Feel free to update to your liking, I just thought the phrasing could use some tweaking.

Reply:
```suggestion
2. If the Agent sets the ReportsHeartbeat capability, the server MUST respond by setting an interval in the heartbeat_interval_seconds field within the OpAMPConnectionSettings message. The value can either be the desired interval, or `0`, indicating that the client should not send heartbeats. 30s is the recommended default interval.
```